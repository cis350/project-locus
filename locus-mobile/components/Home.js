import React from 'react';
import {
  View, Text, StyleSheet, TouchableHighlight,
} from 'react-native';
import Profile from './Profile';

// home page just displays user profile with an option to log out
export default function Home({ route, navigation }) {
  const { user } = route.params;
  function handleLogout() {
    navigation.navigate('Main');
  }
  return (
    <View style={styles.container}>
      <Profile user={user} />
      <TouchableHighlight style={styles.button} onPress={() => handleLogout()} underlayColor="#b00017">
        <Text style={{ textAlign: 'center', fontSize: 30 }}>Logout</Text>
      </TouchableHighlight>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    alignItems: 'center',
  },
  button: {
    backgroundColor: '#ff0022',
    borderRadius: 10,
    paddingVertical: 10,
    width: 150,
    justifyContent: 'center',
    alignItems: 'center',
    marginHorizontal: 10,
  },
});
