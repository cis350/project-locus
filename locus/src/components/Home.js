import React from 'react';
import { getUserFullName } from '../modules/storage';
import Profile from './Profile';

const Home = function HomeComponent({ userEmail }) {
  return (
    <div>
      <Profile userEmail={userEmail} />
    </div>
  );
};

export default Home;
