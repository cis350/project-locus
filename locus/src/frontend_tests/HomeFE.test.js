/**
* @jest-environment jsdom
*/
import React from 'react';
import '@testing-library/jest-dom/extend-expect';

import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Home from '../components/Home';

const mockedNavigator = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedNavigator,
}));

test('member profile test', () => {
  render(<Home userEmail="test@gmail.com" />);
  const textElement = screen.getByText('Loading...');
  expect(textElement).toBeInTheDocument();
});

test('Snapshot Test', () => {
  const component = renderer.create(<Home />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
