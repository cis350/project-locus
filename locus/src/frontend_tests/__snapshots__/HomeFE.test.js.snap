// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<div>
  <div>
    Loading...
  </div>
  <div
    className="d-flex justify-content-center align-content-center "
  >
    <button
      className="btn btn-danger"
      onClick={[Function]}
      style={
        Object {
          "textAlign": "center",
        }
      }
      type="button"
    >
      Logout
    </button>
  </div>
</div>
`;
